src/Authentication/Authenticators/JWT/JWTDecoderInterface.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace CodeIgniter\Shield\Authentication\Authenticators\JWT;
+
+use stdClass;
+
+interface JWTDecoderInterface
```

Comment:
I like the interface in that it allows the JWT Authenticator to exist without the Firebase dependency, because we can pass around the interface name instead of an actual class.

I think the alternative would be to release JWT as a separate "extension" which gets more complex to use and maintain.